Fix inspect with non-primary key id attribute

eugeneius opened the pull request:
Fixes https://github.com/rails/rails/issues/34380.

The `read_attribute` method always returns the primary key when asked to read the `id` attribute, even if the primary key isn't named `id`, and even if another attribute named `id` exists.

For the `inspect`, `attribute_for_inspect` and `pretty_print` methods, this behaviour is undesirable, as they're used to examine the internal state of the record. By using `_read_attribute` instead, we'll get the real value of the `id` attribute.

kamipo commented on activerecord/test/cases/core_test.rb:
```diff
@@ -30,6 +30,11 @@ def test_inspect_limited_select_instance
 
+  def test_inspect_instance_with_non_primary_key_id_attribute
+    topic = topics(:first).becomes(TitlePrimaryKeyTopic)
+    assert_match /id: 1/, topic.inspect
```
Can you address the warning?

```
% ARCONN=sqlite3 be ruby -w -Itest test/cases/core_test.rb
test/cases/core_test.rb:35: warning: ambiguous first argument; put parentheses or a space even after `/' operator
Using sqlite3
Run options: --seed 55120

# Running:

...........

Finished in 0.116315s, 94.5707 runs/s, 180.5441 assertions/s.
11 runs, 21 assertions, 0 failures, 0 errors, 0 skips
```

eugeneius replied:
I fixed it. Thanks for catching that! I forgot that `bin/test` doesn't enable warnings by default 😅

eugeneius committed:
Fix inspect with non-primary key id attribute

The `read_attribute` method always returns the primary key when asked to
read the `id` attribute, even if the primary key isn't named `id`, and
even if another attribute named `id` exists.

For the `inspect`, `attribute_for_inspect` and `pretty_print` methods,
this behaviour is undesirable, as they're used to examine the internal
state of the record. By using `_read_attribute` instead, we'll get the
real value of the `id` attribute.
